### PR TITLE
made changes of form, cards. Many more to change

### PR DESCRIPTION
**Pull request by SayanQi**

### CODE REVIEW

File: app/models/task.rb [hunk 1]
```diff
@@ -1,2 +1,9 @@
 class Task < ApplicationRecord
+  # after_create -> { puts " created this!!!!!!!!!#{self.id}" }
+  after_create :xyz
+
```
Comment (sabyasachi-palit) [inline app/models/task.rb]: add the validations for task name, date should be present. 
And display the error messages accordingly

File: app/assets/config/manifest.js [hunk 2]
```diff
@@ -2,3 +2,5 @@
 //= link_directory ../stylesheets .css
 //= link_tree ../../javascript .js
 //= link_tree ../../../vendor/javascript .js
+//= require moment
+//= require bootstrap-datetimepicker
```
Comment (sabyasachi-palit) [inline app/assets/config/manifest.js]: Check this once, the app is only running after removing moment & bootstrap-datetimepicker

File: app/views/tasks/_task.html.erb [hunk 3]
```diff
@@ -1,16 +1,23 @@
...
+                            <p style="color: green;">Completed</p> 
+                        <% else %>
+                            <p style="color: red;"> Pending </p>
+                        <% end %>
+                    </h6> <!-- Updated class to use Rails Bootstrap helpers -->
                 <p class="card-text"><%= task.description %></p>
-                <%= link_to "Edit", edit_task_path(task), class: "btn btn-secondary", data: { turbo_frame: dom_id(task)} %> 
-                <%= button_to "Delete", method: :delete, form: { data: { turbo_confirm: "Are you sure?" } }, class: "btn btn-danger" %> 
+                <p class="card-text">Due Date : <%= task.pending_date.strftime("%A, %B %d, %Y %I:%M %p") %></p>
+                <div class="d-flex">
```
Comment (sabyasachi-palit) [inline app/views/tasks/_task.html.erb]: In which format you are showing the date?

File: app/views/tasks/_task.html.erb [hunk 4]
```diff
@@ -1,16 +1,23 @@
...
+                </div>
             </div>
         </div>
-        <% end %>
-    </div>
-  
\ No newline at end of file
+    <% end %>
+</div>
+<!-- strftime("%A, %B %d, %Y %I:%M %p") -->
```
Comment (sabyasachi-palit) [inline app/views/tasks/_task.html.erb]: Removed this if not needed

File: app/controllers/tasks_controller.rb [hunk 5]
```diff
@@ -18,35 +23,36 @@ def index
...
-    redirect_to tasks_path
+    redirect_to root_path
    else
     render :edit, status: :unprocessable_entity
    end
   end
 
   def edit
-    @task = Task.find(params[:id])
+    # @task = Task.find(params[:id])
```
Comment (sabyasachi-palit) [inline app/controllers/tasks_controller.rb]: Remove this if not needed

**Review by sabyasachi-palit — CHANGES_REQUESTED** [review]
Add .ruby-gemset file too,

**Comment by SayanQi** [comment]
changes done

File: app/models/task.rb [hunk 6]
```diff
@@ -1,2 +1,11 @@
 class Task < ApplicationRecord
+  # after_create -> { puts " created this!!!!!!!!!#{self.id}" }
+  after_create :xyz
+  validates :name, presence: true
+  validates :description, presence: true
+  validates :pending_date, presence: true
+
+  def xyz
+    puts ("this is xyz metod!!!!!!!")
```
Comment (sabyasachi-palit) [inline app/models/task.rb]: Remove this as its not needed.

File: app/views/layouts/application.html.erb [hunk 7]
```diff
@@ -3,6 +3,7 @@
   <head>
     <title>Todo</title>
     <meta name="viewport" content="width=device-width,initial-scale=1">
+    <meta name="turbo-refresh-method" content="morph">
```
Comment (sabyasachi-palit) [inline app/views/layouts/application.html.erb]: Remove this, it will be not needed I think

File: app/views/tasks/_form.html.erb [hunk 8]
```diff
@@ -1,18 +1,33 @@
...
-        <%= label_tag :completed, "Check me out", class: "form-check-label", for: "exampleCheck1" %>
-    </div>
-    <div class="mb-3">
-            <%= f.button class: "btn btn-success" %>
-        <%= link_to "Cancel", tasks_path, class: "btn btn-secondary" %>
-    </div>
+<% task.errors.each do |key, value| %>
+	<div class="alert alert-<%= key %> alert-dismissible fade show" role="alert">
+			<strong>
+					<%= value %>
```
Comment (sabyasachi-palit) [inline app/views/tasks/_form.html.erb:4]: Please correct the indentation

**Review by sabyasachi-palit — CHANGES_REQUESTED** [review]

**Comment by SayanQi** [comment]
all done

**Review by sabyasachi-palit — APPROVED** [review]

**Pull request merged**